script.js

```diff
@@ -1,4 +1,149 @@
+// below is the random number generator
+var rollDice = function () {
+  // Generate a decimal from 0 through 3, inclusive of 0 and exclusive of 3.
+  var randomDecimal = Math.random() * 3;
+  // Remove the decimal with the floor operation.
+  // This will be an integer from 0 to 2 inclusive.
+  var randomInteger = Math.floor(randomDecimal);
+  // Add 1 to get valid dice rolls of 1 through 3 inclusive.
+  var diceNumber = randomInteger + 1;
+  console.log("this is diceNumber " + diceNumber);
+  return diceNumber;
+};
+// below helps to change the random rolls into a number
+var compChoice = function (diceNumber) {
+  if (diceNumber == 1) {
+    compFinalChoice = "scissors";
+    return "scissors";
+  } else if (diceNumber == 2) {
+    compFinalChoice = "paper";
+    return "paper";
+  } else if (diceNumber == 3) {
+    compFinalChoice = "stone";
+    return "stone";
+  }
+};
+var userGuess = "";
+//below defines the function for the normal game
+var normalGame = function (compChoice, input) {
+  if (
+    (compChoice == "scissors" && input == `scissors`) ||
+    (compChoice == "paper" && input == "paper") ||
+    (compChoice == `stone` && input == `stone`)
+  ) {
+    //below is for the normal game
+    // the below determines who wins if the computer rolls a 1, i.e. scissors
+    myOutputValue = "It's a draw ...";
+    drawCount += 1;
+  } else if (
+    (compChoice == "scissors" && input == "paper") ||
+    (compChoice == "paper" && input == `stone`) ||
+    (compChoice == `stone` && input == `scissors`)
+  ) {
+    myOutputValue = "The computer wins :(";
+    compWins += 1;
+  } else if (
+    (compChoice == "scissors" && input == `stone`) ||
+    (compChoice == "paper" && input == `scissors`) ||
+    (compChoice == `stone` && input == "paper")
+  ) {
+    myOutputValue = "You win!";
+    userWins += 1;
+    // the below determines who wins if the computer rolls a 2, i.e. paper
+  } else {
+    myOutputValue = `${userName}, you have entered something that isn't recognized! Please enter either 'scissors', 'paper', or 'stone'.`;
+  }
+};
+
+var reverseGame = function (compChoice, input) {
+  // if reverse game is true
+  if (
+    (compChoice == "scissors" && input == `reversed scissors`) ||
+    (compChoice == "paper" && input == "reversed paper") ||
+    (compChoice == `stone` && input == `reversed stone`)
+  ) {
+    //below is for the reverse game
+    // the below determines who wins if the computer rolls a 1, i.e. scissors
+    myOutputValue = "It's a draw ...";
+    drawCount += 1;
+  } else if (
+    (compChoice == "scissors" && input == "reversed paper") ||
+    (compChoice == "paper" && input == `reversed stone`) ||
+    (compChoice == `stone` && input == `reversed scissors`)
+  ) {
+    myOutputValue = "You win!";
+    userWins += 1;
+  } else if (
+    (compChoice == "scissors" && input == `reversed stone`) ||
+    (compChoice == "paper" && input == `reversed scissors`) ||
+    (compChoice == `stone` && input == "reversed paper")
```

Comment:
using "reversed paper" is an interesting way of dealing with the reversed game, didn't think of that!